test/e2e/common/pods.go

```diff
@@ -192,8 +191,6 @@ var _ = framework.KubeDescribe("Pods", func() {
 			LabelSelector:   selector.String(),
 			ResourceVersion: pods.ListMeta.ResourceVersion,
 		}
-		w, err := podClient.Watch(options)
```

Comment:
rather than removing the watch, I think we should switch it to `NewIndexerInformerWatcher` to make it more robust

Reply:
@liggitt `NewIndexerInformerWatcher` method is using watch internally, can we avoid flaky by using `NewIndexerInformerWatcher'`.

Reply:
> can we avoid flaky by using `NewIndexerInformerWatcher`

yes

Reply:
@liggitt `NewIndexerInformerWatcher` expects secret object to run the anonymous go routine in 
`k8s.io/client-go/tools/watch/informerwatcher.go`

```
go func() {
		defer close(doneCh)
		informer.Run(w.StopChan())
	}()
```


The error message got from code :

Error message :
```
E0218 16:08:28.560197   24199 reflector.go:348] k8s.io/client-go/tools/watch/informerwatcher.go:113: expected type *v1.Secret, but watch event object had type *v1.Pod
E0218 16:08:28.560235   24199 reflector.go:348] k8s.io/client-go/tools/watch/informerwatcher.go:113: expected type *v1.Secret, but watch event object had type *v1.Pod
E0218 16:08:28.561378   24199 reflector.go:348] k8s.io/client-go/tools/watch/informerwatcher.go:113: expected type *v1.Secret, but watch event object had type *v1.Pod
E0218 16:08:28.562472   24199 reflector.go:348] k8s.io/client-go/tools/watch/informerwatcher.go:113: expected type *v1.Secret, but watch event object had type *v1.Pod
E0218 16:08:28.563600   24199 reflector.go:348] k8s.io/client-go/tools/watch/informerwatcher.go:113: expected type *v1.Secret, but watch event object had type *v1.Pod
E0218 16:08:28.564756   24199 reflector.go:348] k8s.io/client-go/tools/watch/informerwatcher.go:113: expected type *v1.Secret, but watch event object had type *v1.Pod
E0218 16:08:28.565931   24199 reflector.go:348] k8s.io/client-go/tools/watch/informerwatcher.go:113: expected type *v1.Secret, but watch event object had type *v1.Pod
E0218 16:08:28.567040   24199 reflector.go:348] k8s.io/client-go/tools/watch/informerwatcher.go:113: expected type *v1.Secret, but watch event object had type *v1.Pod
Feb 18 16:11:28.553: INFO: timed out waiting for pod deletion
```


Test code :

```
lw := &cache.ListWatch{
			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
				options.LabelSelector = selector.String()
				options.ResourceVersion = pods.ListMeta.ResourceVersion
				return podClient.List(options)
			},
			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
				options.LabelSelector = selector.String()
				options.ResourceVersion = pods.ListMeta.ResourceVersion
				return podClient.Watch(options)
			},
		}
		_, _, w1, done := watchtools.NewIndexerInformerWatcher(lw, &corev1.Secret{})
```

Reply:
> `NewIndexerInformerWatcher` expects secret object

You need to pass the correct type into NewIndexerInformerWatcher, like this:

> _, _, w1, done := watchtools.NewIndexerInformerWatcher(lw, &corev1.Pod{})